ppa/archive/forms.py

```diff
@@ -53,6 +114,45 @@ def set_choices_from_facets(self, facets):
                     (val, mark_safe('%s <span>%d</span>' % (val, count)))
                     for val, count in facet_dict.items()]
 
+    def __init__(self, *args, **kwargs):
+        super(SearchForm, self).__init__(*args, **kwargs)
+
+        pubdate_range = self.pub_date_minmax()
+        # because pubdate is a multifield/multiwidget, access the widgets
+        # under the multiwidgets
+        pubdate_widgets = self.fields['pub_date'].widget.widgets
+        for idx, val in enumerate(pubdate_range):
+            # don't set None as placeholder (only possible if db is empty)
+            if val:
+                # set placeholder and max/min values
+                pubdate_widgets[idx].attrs.update({'placeholder': val,
+                    'min': pubdate_range[0], 'max': pubdate_range[1]})
+            # TODO: add validation! max >= min
+
+    PUBDATE_CACHE_KEY = 'digitizedwork_pubdate_maxmin'
+
+    def pub_date_minmax(self):
+        '''Get minimum and maximum values for
+        :class:`~ppa.archive.models.DigitizedWork` publication dates
+        in the database.  Used to set placeholder values for the form
+        input and to generate the Solr facet range query.
+        Value is cached to avoid repeatedly calculating it.
+
+        :returns: tuple of min, max
+        '''
+        # maxmin = cache.get(self.PUBDATE_CACHE_KEY)
```

Comment:
Am I looking at this wrongly or would this line being commented out result in `pub_date_minmax` never pulling in the value from cache?

Reply:
@bwhicks whoa, thanks for catching that.  I must have commented it out when I was testing some error handling.